test.js

```diff
@@ -98,3 +103,28 @@ test('is.either', t => {
   t.is(is('hi', is.either(is.boolean, is.string), 'hi'), 'hi');
   t.throws(() => is(42, is.either(is.boolean, is.string), '42'));
 });
+
+test('is.literal', t => {
+  t.is(is('true', is.literal('true'), 'true'), 'true');
+  t.throws(() => is(42, is.literal('true'), '42'));
+  /*::
+  let literalAssertion = is.literal<'true'>('true');
+  type LiteralType = AssertionType<typeof literalAssertion>;
+  ('true': LiteralType);
+  */
+});
+
+test('is.literals', t => {
+  let literalsAssertion = is.literals/*::<'a'|'b'|'c'>*/([
+    'a', 'b', 'c'
+  ]);
```

Comment:
I would rather do:

```js
is.oneOf([
  is.literal('default'),
  is.number,
  is.arrayOf(is.number),
])
```

[`$TupleMap`](https://flow.org/en/docs/types/utilities/#toc-tuplemap) will help here

Reply:
Thanks for the tip. I tried it 

```js
let oneOf = /*:: <T: Array<Assertion<any>>> */ (assertions /*:T */) /*: Assertion<$TupleMap<T, ExtractAssertionType>> */ => {
```

But no luck

```
string literal b [1] is incompatible with string literal a [2].

     132│ test('is.oneOf', t => {
     133│   let literalsAssertion = is.oneOf([
 [2] 134│     is.literal/*:: <'a'> */('a'),
 [1] 135│     is.literal/*:: <'b'> */('b'),
     136│     is.literal/*:: <'c'> */('c')
     137│   ]);
     138│   t.is(is('a', literalsAssertion, 'a'), 'a');
```